specification/webpubsub/data-plane/readme.md

```diff
@@ -0,0 +1,158 @@
+# Web PubSub Service
+
+> see https://aka.ms/autorest
+
+This is the AutoRest configuration file for Web PubSub Service.
+
+
+
+---
+## Getting Started
+To build the SDK for Web PubSub Service, simply [Install AutoRest](https://aka.ms/autorest/install) and in this folder, run:
+
+> `autorest`
+
+To see additional help and options, run:
+
+> `autorest --help`
+---
+
+## Configuration
+
+
+
+### Basic Information
+These are the global settings for the Web PubSub Service API.
+
+``` yaml
+openapi-type: data-plane
+tag: package-2021-05-01-preview
+```
+
+### Suppression
+
+``` yaml
+directive:
+  - suppress: LROStatusCodesReturnTypeSchema
+    reason: For this data plane API, it is not a long run operation and the status code indicates the results.
+  - suppress: XmsExamplesRequired
+    reason: There are a lot of APIs that does not have the example. While it is being worked upon disabling this to ensure that we catch and fix other violations
+```
+
+### Tag: package-2021-05-01-preview
+
+These settings apply only when `--tag=package-2021-05-01-preview` is specified on the command line.
+
+``` yaml $(tag) == 'package-2021-05-01-preview'
+input-file:
+- WebPubSub/preview/2021-05-01-preview/webpubsub.json
+title: AzureMessagingWebPubSubServiceClient
+```
+
+# Code Generation
+
+
+## Swagger to SDK
+
+This section describes what SDK should be generated by the automatic system.
+This is not used by Autorest itself.
+
+``` yaml $(swagger-to-sdk)
+swagger-to-sdk:
+  - repo: azure-sdk-for-python
+  - repo: azure-sdk-for-net
+  - repo: azure-sdk-for-java
+  - repo: azure-sdk-for-js
+```
+
+## C#
+
+These settings apply only when `--csharp` is specified on the command line.
+Please also specify `--csharp-sdks-folder=<path to "SDKs" directory of your azure-sdk-for-net clone>`.
+
+``` yaml $(csharp)
+csharp:
+  # last generated with AutoRest.0.17.3
+  azure-arm: false
+  license-header: MICROSOFT_MIT_NO_VERSION
+  namespace: Azure.Messaging.WebPubSub
+  output-folder: $(csharp-sdks-folder)/webpubsub/Azure.Messaging.WebPubSub/src/Generated
+  clear-output-folder: true
+```
+
+## Java
+
+These settings apply only when `--java` is specified on the command line.
+Please also specify `--azure-libraries-for-java-folder=<path to the root directory of your azure-libraries-for-java clone>`.
+
+``` yaml $(java)
+azure-arm: false
+fluent: true
+namespace: com.azure.messaging.webpubsub
+license-header: MICROSOFT_MIT_NO_CODEGEN
+payload-flattening-threshold: 1
+output-folder: $(azure-libraries-for-java-folder)/webpubsub/data-plane
+```
+
+## TypeScript
+
+These settings apply only when `--typescript` is specified on the command line.
+Please also specify `--typescript-sdks-folder=<path to root folder of your azure-sdk-for-js clone>`.
+
+``` yaml $(typescript)
+typescript:
+  azure-arm: false
+  package-name: "@azure/webpubsub"
+  output-folder: "$(typescript-sdks-folder)/sdk/webpubsub/webpubsub"
+  clear-output-folder: true
+  generate-metadata: true
```

Comment:
Why do we have this? Is it for data plane or mgmt plane? If it's for mgmt plane, we already have a readme.typescript.md under resource-manager folder.

Reply:
it's for data plane, any guidance for that?

Reply:
@sarangan12 Can you please help in figuring out why the generation is failing for JS?

Reply:
@vicancy 

Could you please replace the above lines with the following lines:

```
## TypeScript

These settings apply only when `--typescript` is specified on the command line.
Please also specify `--typescript-sdks-folder=<path to root folder of your azure-sdk-for-js clone>`.

``` yaml $(typescript)
azure-arm: false
package-name: "@azure/web-pubsub"
license-header: MICROSOFT_MIT_NO_VERSION
output-folder: "$(typescript-sdks-folder)/sdk/web-pubsub/web-pubsub"
clear-output-folder: true
generate-metadata: true
v3: true
use-extension:
  "@autorest/typescript": "6.0.0-dev.20201105.2"
```

This will ensure that the generated TS SDK use the latest V6 SDK generator instead of standard V4 generator. This will not solve the issue with the CI job (which uses the V4 generator by default) But, we can override that failure and merge in the changes. 

```